frontend/src/app/components/experiments/experiment-template-detail/experiment-template-detail.component.ts

```diff
@@ -11,23 +14,107 @@ import { Observable } from 'rxjs';
   styleUrls: ['./experiment-template-detail.component.scss']
 })
 export class ExperimentTemplateDetailComponent {
-  experimentTemplate$: Observable<ExperimentTemplate>;
+  experimentTemplate: ExperimentTemplate;
+  templateId: string;
+  existExperiments: boolean = false;
   
   constructor(
     protected backend: BackendApiService,
     protected route: ActivatedRoute,
-    protected router: Router
+    protected router: Router,
+    protected dialog: MatDialog
   ) { }
 
   displayedEnvVarColumns: string[] = ['name', 'description'];
 
   ngOnInit() {
     this.route.params.subscribe(params => {
-      this.experimentTemplate$ = this.backend.getExperimentTemplate(params["id"]);
+      this.templateId = params["id"];
+
+      firstValueFrom(this.backend.getExperimentTemplate(params["id"]))
+        .then(template => this.experimentTemplate = template)
+        .catch(err => console.error(err));
+      firstValueFrom(this.backend.getExperimentsOfTeplateCount(params["id"], false))
+        .then(count => this.existExperiments = count > 0)
+        .catch(err => console.error(err));
     });
   }
 
-  getBaseDockerImage(dockerfile: string) {
-    return dockerfile.split("\n")[0].split(" ")[1]
+  editBtnClicked() {
+    let routeParts = ['update'];
+    let routeExtras = { 
+      relativeTo: this.route,
+    };
+
+    if (this.existExperiments) {
+      let str = (
+        "Since there exist Experiments that utilize this particular template, " + 
```

Comment:
"This Experiment Template is already utilized by some existing Experiment. Modifying parameters that could change the template's behavior is restricted."